referenceframe/frame.go

```diff
@@ -445,3 +445,96 @@ func (rf rotationalFrame) MarshalJSON() ([]byte, error) {
 
 	return json.Marshal(temp)
 }
+
+type sevenDFrame struct {
+	*baseFrame
+	collisionGeometry spatial.Geometry
+}
+
+// New7DFrame created a frame with 7 degrees of freedom.
+func New7DFrame(name string, limits []Limit, collisionGeometry spatial.Geometry) (Frame, error) {
+	if len(limits) != 7 {
+		return nil,
+			errors.Errorf("Must have 7DOF state (x, y, z, OX, OY, OZ, theta) to create 6DFrame, have %d dof", len(limits))
+	}
+
+	return &sevenDFrame{
+		&baseFrame{name, limits},
+		collisionGeometry,
+	}, nil
+}
+
+// Transform takes a model and a list of joint angles in radians and computes the dual quaternion representing the
+// cartesian position of the end effector. This is useful for when conversions between quaternions and OV are not needed.
+func (m *sevenDFrame) Transform(inputs []Input) (spatial.Pose, error) {
+	if len(inputs) != 7 {
```

Comment:
Convention is to use the `baseFrame` as the single point of truth so calling `bf.validInputs` here instead